packages/tools/client-debugger/client-debugger-chrome-extension/src/react-components/TelemetryView.tsx

```diff
@@ -31,10 +32,12 @@ export function TelemetryView(): React.ReactElement {
 		/**
 		 * Handlers for inbound messages related to telemetry.
 		 */
+
+		// incoming telemetry event handler
 		const inboundMessageHandlers: InboundHandlers = {
 			["TELEMETRY_EVENT"]: (untypedMessage) => {
 				const message: TelemetryEventMessage = untypedMessage as TelemetryEventMessage;
-				setTelemetryEvents([message.data.contents, ...telemetryEvents]);
+				setTelemetryEvents([...message.data.contents, ...telemetryEvents.reverse()]);
```

Comment:
@chentong7 we actually realized there's a bit of an issue with this line, let's use something like [this](https://github.com/microsoft/FluidFramework/pull/14628/files#diff-85509453c0c6555b93f75aecabee373a04f40cd3466f61fb9c6dd4040ae55a4cR50). Passing a function to `setTelemetryEvents` that can construct the new state based on the old one, so the effect doesn't have to declare `telemetryEvents` as a dependency.

Also, the `reverse()` probably only needs to happen when we receive the list at the beginning; then as we receive new events they can just go at the beginning of the list but we don't need to keep reversing it every time (it'll make it so events are out of order 50% of the time). Or even better... how about we make the `_telemetryLog` insert newer events at the top from the start, so no one has to reverse it? :)

Reply:
Make sense! Updated!